Source/WebKit/GPUProcess/media/RemoteMediaSourceProxy.cpp

```diff
@@ -186,6 +183,15 @@ void RemoteMediaSourceProxy::shutdown()
     m_connectionToWebProcess->connection().sendWithAsyncReply(Messages::MediaSourcePrivateRemote::MediaSourcePrivateShuttingDown(), [self = RefPtr { this }] { }, m_identifier);
 }
 
+void RemoteMediaSourceProxy::invalidate()
+{
+    if (!m_connectionToWebProcess)
+        return;
+
+    m_connectionToWebProcess->messageReceiverMap().removeMessageReceiver(Messages::RemoteMediaSourceProxy::messageReceiverName(), m_identifier.toUInt64());
+    m_connectionToWebProcess = nullptr;
```

Comment:
I think this should be done in `shutdown` instead rather than add yet another method to do the same.

Reply:
by setting m_connectionToWebProcess to nullptr in shutdown.

Reply:
Sure, I extended the lifetime of GPUConnectionToWebProcess so that shutdown can deregister in the reply handler.